### PR TITLE
Show a notification after a successful update

### PR DESCRIPTION
**Pull request by peppy**
Allows access to the github changelog.

### CODE REVIEW

File: osu.Desktop/Overlays/VersionManager.cs [hunk 1]
```diff
@@ -91,6 +99,44 @@ private void load(NotificationOverlay notification, OsuColour colours, TextureSt
...
 
+        protected override void LoadComplete()
+        {
+            base.LoadComplete();
+
+            State = Visibility.Visible;
+
+            var version = game.Version;
+            var lastVersion = config.Get<string>(OsuSetting.Version);
+            if (char.IsNumber(version[0]) && version != lastVersion)
```
Comment (smoogipooo) [inline osu.Desktop/Overlays/VersionManager.cs]: How about just using game.IsDeployedBuild?

File: osu.Desktop/Overlays/VersionManager.cs [hunk 2]
```diff
@@ -91,6 +99,44 @@ private void load(NotificationOverlay notification, OsuColour colours, TextureSt
...
+            {
+                config.Set(OsuSetting.Version, version);
+
+                // only show a notification if we've previously saved a version to the config file (ie. not the first run).
+                if (!string.IsNullOrEmpty(lastVersion))
```
Comment (smoogipooo) [inline osu.Desktop/Overlays/VersionManager.cs]: Can we move this conditional outside of this if? SOmething like
```
if (string.IsNullOrEmpty(lastVersion))
    return;

if (game.IsDeployedBuild && version != lastVersion) { /* etc */ }
```

File: osu.Desktop/Overlays/VersionManager.cs [hunk 3]
```diff
@@ -19,25 +20,32 @@
...
             notificationOverlay = notification;
+            this.config = config;
+            this.game = game;
 
             AutoSizeAxes = Axes.Both;
             Anchor = Anchor.BottomCentre;
             Origin = Anchor.BottomCentre;
+
             Alpha = 0;
+            State = Visibility.Hidden;
```
Comment (smoogipooo) [inline osu.Desktop/Overlays/VersionManager.cs]: What's the point of setting state to hidden if it's just going to be set to visible in loadcomplete?

File: osu.Desktop/OsuGameDesktop.cs [hunk 4]
```diff
@@ -88,11 +83,15 @@ protected override void LoadComplete()
...
-            LoadComponentAsync(versionManager, Add);
+            LoadComponentAsync(versionManager = new VersionManager { Depth = int.MinValue });
+
             ScreenChanged += s =>
             {
-                if (!versionManager.IsPresent && s is Intro)
+                if (s is Intro && s.ChildScreen == null)
+                {
+                    Add(versionManager);
                     versionManager.State = Visibility.Visible;
```
Comment (smoogipooo) [inline osu.Desktop/OsuGameDesktop.cs]: Again, is there any point to setting visibility here given that it's going to be visible in loadcomplete?

File: osu.Game/Overlays/Notifications/SimpleNotification.cs [hunk 5]
```diff
@@ -4,17 +4,16 @@
...
 using osu.Game.Graphics;
-using osu.Game.Graphics.Sprites;
 using OpenTK;
 
 namespace osu.Game.Overlays.Notifications
 {
     public class SimpleNotification : Notification
     {
-        private string text;
+        private string text = string.Empty;
```
Comment (smoogipooo) [inline osu.Game/Overlays/Notifications/SimpleNotification.cs]: This is the default value.

**Review by smoogipoo — CHANGES_REQUESTED** [review]

File: osu.Game/Overlays/Notifications/SimpleNotification.cs (hunk 5 above)
Comment (peppy) [inline osu.Game/Overlays/Notifications/SimpleNotification.cs]: default is null; causes a nullref

File: osu.Desktop/Overlays/VersionManager.cs (hunk 3 above)
Comment (peppy) [inline osu.Desktop/Overlays/VersionManager.cs]: actually hidden is the default state, so this is redundant.

File: osu.Desktop/Overlays/VersionManager.cs (hunk 1 above)
Comment (peppy) [inline osu.Desktop/Overlays/VersionManager.cs]: good point

File: osu.Game/Overlays/Notifications/SimpleNotification.cs (hunk 5 above)
Comment (smoogipooo) [inline osu.Game/Overlays/Notifications/SimpleNotification.cs]: whoah. ok

File: osu.Desktop/OsuGameDesktop.cs (hunk 4 above)
Comment (peppy) [inline osu.Desktop/OsuGameDesktop.cs]: yep, one of these is redundant. i left too many in (found a pretty serious bug with VisibleContainer - see framework PR)

File: osu.Desktop/Overlays/VersionManager.cs (hunk 2 above)
Comment (peppy) [inline osu.Desktop/Overlays/VersionManager.cs]: can do

Comment (peppy) [inline osu.Desktop/Overlays/VersionManager.cs]: can't do; the set to config needs to only happen under the conditionals it is contained by

Comment (smoogipooo) [inline osu.Desktop/Overlays/VersionManager.cs]: Right.

**Review by smoogipoo — APPROVED** [review]

**Pull request merged**